pypeit/images/rawimage.py

```diff
@@ -223,10 +223,14 @@ def process(self, process_steps, par, bpm=bpm, flatimages=None, bias=None,
             self.spat_flexure_shift = flexure.spat_flexure_shift(self.image, slits)
 
         # Generate the illumination flat
+        illum_flat = None
         if self.par['illumflatten']:
-            if flatimages is None or slits is None:
-                msgs.error("Need to provide slits and flatimages to illumination flat")
-            illum_flat = flatimages.fit2illumflat(slits, flexure_shift=self.spat_flexure_shift)
+            if flatimages is None:
+                msgs.warn("Cannot illumflatten, no flat field image generated. Skipping..")
+            if slits is None:
```

Comment:
Do you want this to be `elif`?  I.e., what happens if `flatimages` is None, but `slits` is not?

Rest looks fine.

Reply:
yup, good catch..